### PR TITLE
Bad log message

### LINKED ISSUE #27695: Bad log message
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

A log message in FrontController.php send "Cart cannot be loaded..." while the tests made to get there do not test this possibility.

### Expected behavior

Change the text of the log or really do the test.

### Steps to reproduce

Redo an order with the same cart.

### PrestaShop version(s) where the bug happened

All

### PHP version(s) where the bug happened

_No response_

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by glv-soledis**
The log message does not correspond to the tests previously performed.

|Questions      |Answers
|---------------|----------------------------------------------
|Branch?        |develop
|Description?   |The log message does not correspond to the tests previously performed.
|Type?          |improvement
|Category?      |BO
|BC breaks?     |no
|Deprecations?  |no
|Fixed ticket?  |Fixes #27695
|How to test?   |Redo an order with the same cart.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27694)

### CODE REVIEW

**Comment by glv-soledis** [comment]
|Questions      |Answers
|---------------|----------------------------------------------
|Branch?        |develop
|Description?   |The log message does not correspond to the tests previously performed.
|Type?          |improvement
|Category?      |BO
|BC breaks?     |no
|Deprecations?  |no
|Fixed ticket?  |Fixes #27695
|How to test?   |Redo an order with the same cart.

File: classes/controller/FrontController.php [hunk 1]
```diff
@@ -393,11 +393,11 @@ public function init()
 
         /* Cart already exists */
         if ((int) $this->context->cookie->id_cart) {
-            if (!isset($cart)) {
+            if (!isset($cart) || !Validate::isLoadedObject($cart)) {
                 $cart = new Cart($this->context->cookie->id_cart);
             }
 
-            if (Validate::isLoadedObject($cart) && $cart->orderExists()) {
+            if (!Validate::isLoadedObject($cart) || $cart->orderExists()) {
```
Comment (matks) [inline classes/controller/FrontController.php:400]: Indeed this seems more logical 😄

File: classes/controller/FrontController.php [hunk 2]
```diff
@@ -393,11 +393,11 @@ public function init()
 
         /* Cart already exists */
         if ((int) $this->context->cookie->id_cart) {
-            if (!isset($cart)) {
+            if (!isset($cart) || !Validate::isLoadedObject($cart)) {
```
Comment (matks) [inline classes/controller/FrontController.php:396]: So we re-create the cart object if the previous cart object cannot be loaded

I'm a little worried that then we would load some cart content 🤔  do you know how this usecase can happen?

**Review by matks — CHANGES_REQUESTED** [review]
Dear @glv-soledis thank you for the contribution however we need the bug report https://github.com/PrestaShop/PrestaShop/issues/27695 to be confirmed so we can go forward.

We can only validate your PR if we have a clear, reproducible scenario we can follow to observe the bad behavior. In order to be able to review your PR, to verify any side-effects, to make sure it is correct, we need a complete understanding of it, and so does the QA team. That's why a list of steps "1. do this 2. do then that 3. we should see ... but instead we should see ..." is a must-have needed for us to verify the bug 😉 .

File: classes/controller/FrontController.php (hunk 2 above)
Comment (glv-soledis) [reply classes/controller/FrontController.php:396]: For me it is mainly a security in case the global $cart is upset.

**Comment by ghost** [comment]
Hello,
You will find below the steps to display the error messages in the backoffice.

1. Have 2 tabs open on the "shopping cart ?action=show" page (with items)
image

1. Launch and complete the order on one of the tabs
image

1. Refresh the other tab still active on the old cart 

1. Go to the back-office in "Advanced settings --> Logs" and the following error message should appear 
image

**Comment by Progi1984** [comment]
Closed in favor of #27839 by @tlgsoledis (I think it's a team member of Soledis :muscle:).

**Pull request closed without merging**